src/common/configuration/file-system-configuration.ts

```diff
@@ -29,6 +29,8 @@ export class FileSystemConfiguration implements ConfigAccessor, ConfigMutator {
 
     public reset(): ConfigMutator {
         try {
+            // The following warning is disabled because this.configJsonPath is a string literal whose value is set differently only for unit tests
+            // eslint-disable-next-line security/detect-non-literal-fs-filename
             const configJsonRawContent = fs.readFileSync(this.configJsonPath).toString();
```

Comment:
I think this is actually somewhat legitimate; even if it isn't a problem on its own because we only use it intentionally with a constant parameter, it would allow a bad actor a path for escalate "property pollution of the config object" into "arbitrary file read".

If the path override is only used by unit tests anyway, probably the easiest fix is to just inject `fs.readFileSync` as the dependency in place of `configJsonPath` and update the tests replacing `testFilePath` parameters with either `() => fs.readFileSync(testFilePath)` or `() => Buffer.from(inlineTestFileContent)`. (I'd go with the former despite it being a unit test that touches the file system, better test of the real error handling responsibilities of the class)